### PR TITLE
Initial implementation of mapbox::base::Value

### PR DESCRIPTION
**Pull request by pozdnyakov**

### CODE REVIEW

File: mapbox/CMakeLists.txt [hunk 1]
```diff
@@ -34,5 +34,6 @@ mapbox_base_add_library(supercluster.hpp ${CMAKE_CURRENT_LIST_DIR}/supercluster.
 mapbox_base_add_library(variant ${CMAKE_CURRENT_LIST_DIR}/variant/include)
 mapbox_base_add_library(weak ${CMAKE_CURRENT_LIST_DIR}/weak/include)
 mapbox_base_add_library(typewrapper ${CMAKE_CURRENT_LIST_DIR}/typewrapper/include)
-
+mapbox_base_add_library(value ${CMAKE_CURRENT_LIST_DIR}/value/include)
+target_link_libraries(mapbox-base-value INTERFACE mapbox-base-variant)
```
Comment (brunoabinader) [inline mapbox/CMakeLists.txt:38]: :+1:

File: test/value.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    checkType<uint64_t>(32u);
+    checkType<double>(32.0);
+    checkType<bool>(false);
+    checkType<std::string>("hello");
+
+    mapbox::base::Value intValue{32};
+    try {
+        intValue.get<uint64_t>();
+        assert(false); // Should throw.
+    } catch (...) {
```
Comment (brunoabinader) [inline test/value.cpp:29]: Isn't the `assert(false)` supposed to be inside the catch scope?

**Comment by brunoabinader** [comment]
@pozdnyakov After some thought on this, I don't think `mapbox-base` is the right place for `Value`. Instead, I think we should move this implementation to [geometry.hpp](https://github.com/mapbox/geometry.hpp) and replace the current [mapbox::feature::value](https://github.com/mapbox/geometry.hpp/blob/master/include/mapbox/feature.hpp#L36) with this implementation.

File: test/value.cpp (hunk 2 above)
Comment (pozdnyakov) [reply test/value.cpp:29]: nope, it should throw: it's `int64_t` not `uint64_t`

**Comment by pozdnyakov** [comment]
>  I don't think `mapbox-base` is the right place for Value. Instead, I think we should move this implementation to `geometry.hpp` and replace the current `mapbox::feature::value` with this implementation.

Let's instead make `geometry.hpp` use `mapbox::base::Value` :smile:

**Comment by pozdnyakov** [comment]
Agreed to use `mapbox::feature::value` as a short-term solution.

**Pull request closed without merging**